From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Tue, 11 Oct 2011 08:36:44 -0600
Patch-mainline: v3.2-rc1
Subject: [PATCH] Staging: hv: remove unneeded asm include file in hyperv.h
Git-commit: 407dd1644302ea78fa5d740e67a1c09677aa18a4

No one outside of the hyperv core needs to include the asm/hyperv.h
file, so don't put it in the "global" include/linux/hyperv.h file.

Cc: K. Y. Srinivasan <kys@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/hv/connection.c |    2 +-
 drivers/hv/hv.c         |    2 +-
 drivers/hv/vmbus_drv.c  |    2 +-
 include/linux/hyperv.h  |    3 ---
 4 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/hv/connection.c b/drivers/hv/connection.c
index 5f438b6..650c9f0 100644
--- a/drivers/hv/connection.c
+++ b/drivers/hv/connection.c
@@ -30,7 +30,7 @@
 #include <linux/slab.h>
 #include <linux/vmalloc.h>
 #include <linux/hyperv.h>
-
+#include <asm/hyperv.h>
 #include "hyperv_vmbus.h"
 
 
diff --git a/drivers/hv/hv.c b/drivers/hv/hv.c
index 931b7b0..0fb100e 100644
--- a/drivers/hv/hv.c
+++ b/drivers/hv/hv.c
@@ -26,7 +26,7 @@
 #include <linux/slab.h>
 #include <linux/vmalloc.h>
 #include <linux/hyperv.h>
-
+#include <asm/hyperv.h>
 #include "hyperv_vmbus.h"
 
 /* The one and only */
diff --git a/drivers/hv/vmbus_drv.c b/drivers/hv/vmbus_drv.c
index b0d08f9..bb9f205 100644
--- a/drivers/hv/vmbus_drv.c
+++ b/drivers/hv/vmbus_drv.c
@@ -33,7 +33,7 @@
 #include <acpi/acpi_bus.h>
 #include <linux/completion.h>
 #include <linux/hyperv.h>
-
+#include <asm/hyperv.h>
 #include "hyperv_vmbus.h"
 
 
diff --git a/include/linux/hyperv.h b/include/linux/hyperv.h
index edaa9e2..4c8414a 100644
--- a/include/linux/hyperv.h
+++ b/include/linux/hyperv.h
@@ -35,9 +35,6 @@
 #include <linux/mod_devicetable.h>
 
 
-#include <asm/hyperv.h>
-
-
 #define MAX_PAGE_BUFFER_COUNT				16
 #define MAX_MULTIPAGE_BUFFER_COUNT			32 /* 128K */
 
-- 
1.6.0.2


